lib/runtime/src/lib.rs

```diff
@@ -108,8 +108,8 @@ impl SystemHealth {
         self.system_health = status;
     }
 
-    pub fn set_endpoint_health_status(&mut self, endpoint: String, status: HealthStatus) {
-        self.endpoint_health.insert(endpoint, status);
+    pub fn set_endpoint_health_status(&mut self, endpoint: &str, status: HealthStatus) {
+        self.endpoint_health.insert(endpoint.to_string(), status);
```

Comment:
nit, but there's no need to change this. The previous version allowed for if the caller already owns a `String`, and is giving it to you. No allocation. Here you are forcing the allocation.

Reply:
went back and forth on this as well - I wanted to avoid cloning the endpoint name on every request (for the tracing code) and so made this a reference to map it - my understanding is that in the insert() it will be copied regardless? as the String is not cloneable - may be missing the nuance here - my thought was to wait until the copy is needed which I think happens no matter what at the insert() 

but is the point here that it could be optimized out by the compiler in the original?

Reply:
`String::clone` and `&str::to_string` do the same thing, allocate new memory and copy the contents into it, so I guess it works out the same either way. No worries.